shared/constants/chat2/message.js

```diff
@@ -972,7 +972,25 @@ const imageFileNameRegex = /[^/]+\.(jpg|png|gif|jpeg|bmp)$/i
 export const pathToAttachmentType = (path: string) => (imageFileNameRegex.test(path) ? 'image' : 'file')
 export const isSpecialMention = (s: string) => ['here', 'channel', 'everyone'].includes(s)
 
-export const upgradeMessage = (old: Types.Message, m: Types.Message) => {
+export const upgradeMessage = (_old: ?Types.Message, m: Types.Message, merge: boolean) => {
+  if (merge) {
```

Comment:
this function does completely different things depending on `merge` -- maybe they should just be different functions?